include/aktualizr-lite/tuf/tuf.h

```diff
@@ -0,0 +1,142 @@
+#include "uptane/fetcher.h"
+#include "uptane/imagerepository.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+
+namespace Uptane {
+class Target;
+}
+
+namespace aklite {
+namespace tuf {
+
+// enum class RoleEnum { Root = 0, Snapshot = 1, Targets = 2, Timestamp = 3, InvalidRole = -1 };
+// virtual std::string fetchRole(RoleEnum role, int version) = 0;
+
+// TODO: consider switching to a single fetchRole
+class TufRepoSource {
+ public:
+  virtual std::string fetchRoot(int version) = 0;
+  virtual std::string fetchTimestamp() = 0;
+  virtual std::string fetchSnapshot() = 0;
+  virtual std::string fetchTargets() = 0;
+};
+
+// TufRepoSource implementation for fetching remote meta-data via https using libaktualizr
+class AkHttpsTufRepoSource : public TufRepoSource {
+ public:
+  AkHttpsTufRepoSource(std::string name_in, boost::property_tree::ptree& pt);
+  ~AkHttpsTufRepoSource();
+
+  virtual std::string fetchRoot(int version);
+  virtual std::string fetchTimestamp();
+  virtual std::string fetchSnapshot();
+  virtual std::string fetchTargets();
+
+ private:
+  void init(std::string name_in, boost::property_tree::ptree& pt);
+  void fillConfig(Config& config, boost::property_tree::ptree& pt);
+  std::string fetchRole(Uptane::Role role, int64_t maxsize, Uptane::Version version);
+
+  std::string name_;
+  boost::filesystem::path tmp_dir_path_;
+  std::shared_ptr<Uptane::IMetadataFetcher> meta_fetcher_;
+};
+
+// TufRepoSource implementation for fetching local meta-dat
+class AkLocalTufRepoSource : public TufRepoSource {
+ public:
+  class NotFoundException : public std::runtime_error {
+   public:
+    NotFoundException(const std::string& path) : std::runtime_error("Metadata hasn't been found; file path: " + path) {}
+  };
+
+  AkLocalTufRepoSource(std::string name_in, boost::property_tree::ptree pt);
+  AkLocalTufRepoSource(std::string uri);
+  // DISCUSS: limit to max version?
+  virtual std::string fetchRoot(int version);
+  virtual std::string fetchTimestamp();
+  virtual std::string fetchSnapshot();
+  virtual std::string fetchTargets();
+
+ private:
+  virtual std::string fetchFile(boost::filesystem::path meta_file_path);
+
+  std::string name;
+  boost::filesystem::path src_dir_;
+};
+
+// Wrapper around any TufRepoSource implementation to make it usable directly by libaktualizr,
+// by implementing Uptane::IMetadataFetcher interface
+class FetcherWrapper : public Uptane::IMetadataFetcher {
+ public:
+  FetcherWrapper(std::shared_ptr<TufRepoSource> src);
+  void fetchRole(std::string* result, int64_t maxsize, Uptane::RepositoryType repo, const Uptane::Role& role,
+                 Uptane::Version version) const;
+
+  void fetchLatestRole(std::string* result, int64_t maxsize, Uptane::RepositoryType repo,
+                       const Uptane::Role& role) const;
+
+ private:
+  std::shared_ptr<TufRepoSource> repo_src;
+};
+
+// Discuss: Keep custom as Json, or add explicit fields for the required values
+struct Target {
```

Comment:
I think we should try and use this "target" type https://github.com/foundriesio/aktualizr-lite/blob/2afacef575719f83da3b99bf4c3f7226bd09a89d/include/aktualizr-lite/api.h#L22 (`TufTarget`). Specifically, move `TufTarget` from `api.h` to this `tuf.h` and add "include `tuf.h` " to `api.h`

Reply:
Done